### ISSUE TITLE
False red highlight on nightly feature `associated_type_defaults` 

### ISSUE THREAD

**Issue by citizen-stig** [intellij-rust/intellij-rust#10704]
## Environment

* **IntelliJ Rust plugin version: 0.4.198.5409-231**
* **Rust toolchain version: channel = "nightly-2023-07-15" profile = "minimal"**
* **IDE name and version: CLion 2023.1.4 Build #CL-231.9161.40, built on June 20, 2023**
* **Operating system: MacOS Ventura 13.4.1, Intel CPU**

Runtime version: 17.0.7+10-b829.16 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

## Problem description

IntelliJ-Rust plugin incorrectly flags an error when using the nightly feature associated_type_defaults to override an associated default type in a trait.

The problem arises in the following [code snippet](https://github.com/citizen-stig/intellij-rust-demos/blob/a3baee4a5331d0f0b1ccf6ec77e8cf3f106592a6/src/lib.rs)

## Steps to reproduce

1. Begin by creating a trait with an associated default type, as shown below:

```rust
#![feature(associated_type_defaults)]

#[derive(Debug, PartialEq)]
pub enum NonInstantiable {}

pub trait Module {
    type CallMessage: std::fmt::Debug = NonInstantiable;

    fn call(&mut self, _message: Self::CallMessage) -> Result<(), anyhow::Error> {
        unreachable!("call message is not instantiable")
    }
}

pub struct MyModule {
    storage: Vec<u8>,
}

#[derive(Debug, PartialEq)]
pub struct MyCallMessage {
    data: Vec<u8>,
}

impl Module for MyModule {
    type CallMessage = MyCallMessage;

    fn call(&mut self, message: Self::CallMessage) -> Result<(), anyhow::Error> {
        println!("MyModule::call({:?})", message);
        self.storage.extend(&message.data);
        Ok(())
    }
}
```
2. Implement the trait and override the associated default type with a custom type.

### Actual Result

The IntelliJ-Rust plugin throws an error, indicating that an incorrect type is provided for the implementation. Surprisingly, it seems to expect the default type even when it has been overridden:

```
mismatched types [E0308] expected `NonInstantiable`, found `MyCallMessage` 
```

<img width="935" alt="Screenshot 2023-07-17 at 13 43 39" src="https://github.com/intellij-rust/intellij-rust/assets/3115300/40cd07eb-9fba-420a-bd30-9604c1fa69b5">

## Expected Result

The IntelliJ-Rust plugin should correctly recognize and handle the override of the associated default type without any error.

**Comment by Maksim20023** [comment]
Hi! Thank you for your report.
I forwarded your case and steps for reproducing it to our developer's team. It will be helpful for us. If you have any further suggestions, please feel free to reach out.
